File: typescript/medium/minOperations.ts
function minOperations(nums: number[]): number {
  let dp1 = 0;
  let dp2 = 0;
  if (nums[0] == 0) {
    dp1 = 1;
  }
  for (let i = 1; i < nums.length; i++) {
    dp2 = dp1;
    if (nums[i] != nums[i - 1]) {
      dp2++;
    }
    dp1 = dp2;
  }
  return dp1;
}

File: java/src/easy/MinOperations.java
package easy;

public class MinOperations {
    public int minOperations(int[] nums) {
        int left=0;
        int right=2;
        int operators=0;
        while(left<nums.length){
            if(nums[left]==1){
                left++;
                right++;
            }else{
                if(right>=nums.length){
                    return -1;
                }
                operators++;
                for(int i=left;i<=right;i++){
                    nums[i]=nums[i]^1;
                }
            }
        }
        return operators;
    }
}
